<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="elon_musk_car">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="-0.0248489579671259 -1.69844779697725E-05 0.0595015198553598"
        rpy="0 0 0" />
      <mass
        value="1.07304100169638" />
      <inertia
        ixx="0.00185580886331691"
        ixy="3.12605392585196E-07"
        ixz="-5.60262178799118E-05"
        iyy="0.00450498080335361"
        iyz="8.92348407742513E-07"
        izz="0.00573561869820229" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.831372549019608 0.670588235294118 0.129411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="traiduoi">
    <inertial>
      <origin
        xyz="1.12801188736911E-08 -0.00907479627188329 -1.48281719958288E-08"
        rpy="0 0 0" />
      <mass
        value="0.0521078133004598" />
      <inertia
        ixx="3.28190540725205E-05"
        ixy="-9.48347094999047E-13"
        ixz="-9.02704512031813E-10"
        iyy="5.52101334726658E-05"
        iyz="1.24402576673555E-12"
        izz="3.28192928378622E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/traiduoi.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/traiduoi.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="traiduoi_jt"
    type="continuous">
    <origin
      xyz="-0.0842500000002616 0.089625 0.0113156378829189"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="traiduoi" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="phaiduoi">
    <inertial>
      <origin
        xyz="1.25285689461796E-09 0.00907479627188298 1.85888698739367E-08"
        rpy="0 0 0" />
      <mass
        value="0.0521078133004657" />
      <inertia
        ixx="3.28182966727171E-05"
        ixy="1.06769615106998E-13"
        ixz="2.45723743418195E-10"
        iyy="5.5210133472672E-05"
        iyz="1.56062810555671E-12"
        izz="3.28200502376713E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/phaiduoi.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/phaiduoi.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="phaiduoi_jt"
    type="continuous">
    <origin
      xyz="-0.084250000000344 -0.089625 0.0113156378831937"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="phaiduoi" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="traitren">
    <inertial>
      <origin
        xyz="1.55709193594422E-08 -0.00907479627188237 -1.02304585497537E-08"
        rpy="0 0 0" />
      <mass
        value="0.0521078133004729" />
      <inertia
        ixx="3.28196469526073E-05"
        ixy="-1.30821560234278E-12"
        ixz="-7.77771096015315E-10"
        iyy="5.521013347268E-05"
        iyz="8.57635029502247E-13"
        izz="3.28186999577899E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/traitren.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/traitren.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="traitren_jt"
    type="continuous">
    <origin
      xyz="0.0842499999998293 0.089625 0.0113156378828449"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="traitren" />
    <axis
      xyz="0 0.999999731823558 0.00073236112126337" />
  </joint>
  <link
    name="phaitren">
    <inertial>
      <origin
        xyz="1.79235902231234E-08 0.00907479627188346 5.08636011087304E-09"
        rpy="0 0 0" />
      <mass
        value="0.0521078133004571" />
      <inertia
        ixx="3.28200069638325E-05"
        ixy="1.50529068461448E-12"
        ixz="3.66593789169686E-10"
        iyy="5.52101334726631E-05"
        iyz="4.25515881150075E-13"
        izz="3.28183399465464E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/phaitren.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/phaitren.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="phaitren_jt"
    type="continuous">
    <origin
      xyz="0.08425 -0.089625 0.011184"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="phaitren" />
    <axis
      xyz="0 1 0.00073236" />
  </joint>
  <link
    name="truc1">
    <inertial>
      <origin
        xyz="0.0533181945720658 -0.00959003139334669 0.0453560373883681"
        rpy="0 0 0" />
      <mass
        value="0.108511590245375" />
      <inertia
        ixx="0.000127329624285286"
        ixy="-9.49294539450881E-12"
        ixz="-0.000133704438877886"
        iyy="0.000291291060775154"
        iyz="-4.06646426432114E-12"
        izz="0.000170767608899266" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/truc1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.831372549019608 0.670588235294118 0.129411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/truc1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="truc1_jt"
    type="revolute">
    <origin
      xyz="-0.067056 -0.01 0.16393"
      rpy="0 -0.5236 0" />
    <parent
      link="base_link" />
    <child
      link="truc1" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.5"
      upper="1"
      effort="5"
      velocity="1" />
  </joint>
  <link
    name="truc2">
    <inertial>
      <origin
        xyz="0.0573218605022251 0.00973273237026839 -0.0234797350529898"
        rpy="0 0 0" />
      <mass
        value="0.100138675077548" />
      <inertia
        ixx="4.14451577467379E-05"
        ixy="7.90091994280198E-08"
        ixz="7.14211200024037E-05"
        iyy="0.000221871143736014"
        iyz="-3.23784009562632E-08"
        izz="0.000186552928423222" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/truc2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.831372549019608 0.670588235294118 0.129411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/truc2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="truc2_jt"
    type="revolute">
    <origin
      xyz="0.106636389144134 0 0.0907120747767362"
      rpy="0 -0.2618 0" />
    <parent
      link="truc1" />
    <child
      link="truc2" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.5"
      upper="1"
      effort="5"
      velocity="1" />
  </joint>
  <link
    name="imu">
    <inertial>
      <origin
        xyz="-6.68985725750804E-05 0.000170855890153421 -0.00067929222391197"
        rpy="0 0 0" />
      <mass
        value="0.0004529996914439" />
      <inertia
        ixx="1.17080889572944E-08"
        ixy="-5.09802080398037E-15"
        ixz="-3.97332758659873E-16"
        iyy="5.34815547364709E-09"
        iyz="-7.20405912899209E-16"
        izz="1.6875100792001E-08" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/imu.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/imu.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="imu_jt"
    type="fixed">
    <origin
      xyz="-0.12975 0.00015956 0.070527"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="imu" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="lidar">
    <inertial>
      <origin
        xyz="-0.00748346988333637 -9.19261110875202E-05 -0.0297280760651208"
        rpy="0 0 0" />
      <mass
        value="1.33076913637071" />
      <inertia
        ixx="0.00061491794045409"
        ixy="-8.01984318975285E-07"
        ixz="0.00015824302591117"
        iyy="0.000960697335617222"
        iyz="-3.0916298658758E-06"
        izz="0.00109815200027136" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/lidar.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.250980392156863 0.250980392156863 0.250980392156863 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/lidar.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="lidar_jt"
    type="fixed">
    <origin
      xyz="0.05075 0 0.11892"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="lidar" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="camera">
    <inertial>
      <origin
        xyz="-0.0104719793283222 -1.35431902550109E-05 -0.000101038487187566"
        rpy="0 0 0" />
      <mass
        value="0.00511234868100241" />
      <inertia
        ixx="6.22026605833524E-07"
        ixy="-1.40839446300285E-09"
        ixz="-1.19148777131332E-08"
        iyy="4.54335524710268E-07"
        iyz="1.42394873810616E-09"
        izz="6.01477590970195E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/camera.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.901960784313726 0.709803921568627 0.43921568627451 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://elon_musk_car/meshes/camera.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="camera_jt"
    type="fixed">
    <origin
      xyz="0.153828045195748 0 0.0450277243285144"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="camera" />
    <axis
      xyz="0 0 0" />
  </joint>

  <gazebo>
    <plugin name="skid_steer_drive_controller" filename="libgazebo_ros_skid_steer_drive.so">
      <updateRate>100.0</updateRate>
      <robotNamespace>/</robotNamespace>
      <leftFrontJoint>traitren_jt</leftFrontJoint>
      <rightFrontJoint>phaitren_jt</rightFrontJoint>
      <leftRearJoint>traiduoi_jt</leftRearJoint>
      <rightRearJoint>phaiduoi_jt</rightRearJoint>
      <wheelSeparation>0.15</wheelSeparation>      
      <wheelDiameter>0.09</wheelDiameter>       
      <robotBaseFrame>base_link</robotBaseFrame>
      <torque>20</torque>
        <commandTopic>/cmd_vel</commandTopic>
      <broadcastTF>false</broadcastTF>
    </plugin>
  </gazebo>

  <gazebo reference="camera">
    <sensor type="camera" name="camera_sensor">
      <always_on>true</always_on>
      <update_rate>30.0</update_rate>
      <camera>
        <horizontal_fov>1.047</horizontal_fov>
        <image>
          <width>640</width>
          <height>480</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.1</near>
          <far>100</far>
        </clip>
      </camera>
      <plugin name="gazebo_ros_camera" filename="libgazebo_ros_camera.so">
        <cameraName>camera</cameraName>
        <imageTopicName>image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera</frameName>
      </plugin>
    </sensor>
  </gazebo>

  <gazebo reference="imu">
    <sensor type="imu" name="imu_sensor">
      <always_on>true</always_on>
      <update_rate>50.0</update_rate>
      <plugin name="imu_plugin" filename="libgazebo_ros_imu_sensor.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>50.0</updateRate>
        <bodyName>imu</bodyName>
        <topicName>imu/data</topicName>
        <frameName>imu</frameName>
      </plugin>
    </sensor>
  </gazebo>

  <gazebo reference="lidar">
    <sensor type="ray" name="lidar_sensor">
      <pose>0 0 0 0 0 0</pose>
      <visualize>true</visualize>
      <update_rate>30</update_rate>
      <ray>
        <scan>
          <horizontal>
            <samples>720</samples>
            <resolution>1</resolution>
            <min_angle>-1.5708</min_angle>
            <max_angle>1.5708</max_angle>
          </horizontal>
        </scan>
        <range>
          <min>0.1</min>
          <max>30.0</max>
          <resolution>0.01</resolution>
        </range>
        <noise>
          <type>gaussian</type>
          <mean>0.0</mean>
          <stddev>0.01</stddev>
        </noise>
      </ray>
      <plugin name="gazebo_ros_laser" filename="libgazebo_ros_laser.so">
        <topicName>/scan</topicName>
        <frameName>lidar</frameName>
      </plugin>
    </sensor>
  </gazebo>

  <transmission name="truc1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="truc1_jt">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="truc1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="truc2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="truc2_jt">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="truc2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so"/>
  </gazebo>

</robot>